src/StatusQ/Controls/StatusInput.qml

```diff
@@ -0,0 +1,85 @@
+import QtQuick 2.14
+
+import StatusQ.Core 0.1
+import StatusQ.Core.Theme 0.1
+import StatusQ.Controls 0.1
+
+Rectangle {
+    id: root
+
+    implicitWidth: 480
+    height: (label.visible ? 
```

Comment:
Qt Layout system does all this math for you if you were using `ColumnLayout` instead of anchors.
```
FocusScope {
    id: root
    implicitHeight: layout.childrenRect.height

    ColumnLayout {
        id: layout
        
        StatusBaseText {
            Layout.fillWidth: true // for elide to work
        }
        
        StatusBaseInput {
            Layout.fillWidth: true
        } 
        ... etc
    }
}
```

Reply:
Ah, that's cool! I'm very bad at this layout stuff haha